Sources/ProjectDescription/Template.swift

```diff
@@ -0,0 +1,76 @@
+public struct Template: Codable {
+    public let description: String
+    public let attributes: [Attribute]
+    public let files: [File]
+    public let directories: [String]
+    
+    public init(description: String,
+                arguments: [Attribute] = [],
+                files: [File] = [],
+                directories: [String] = []) {
+        self.description = description
+        self.attributes = arguments
+        self.files = files
+        self.directories = directories
+        dumpIfNeeded(self)
+    }
+}
+
+public extension String.StringInterpolation {
```

Comment:
Do you have an example of what this extension is for? I can't see its use case from the implementation.

Reply:
```swift
let testContents = """
// this is some \(nameArgument) content
"""
```
-> I need to encode and then decode contents of the file - and that means encoding and decoding the attribute; this turns it to `{{ attribute }}` for which I can then later fill in the value that the user inputs.

Reply:
What about using the ExpressibleByStringInterpolation protocol like we are doing [here](https://github.com/tuist/tuist/blob/master/Sources/TuistSupport/Utils/Printer.swift#L21)? We [follow](https://github.com/tuist/tuist/blob/master/Sources/ProjectDescription/TuistConfig.swift#L9) a similar approach to define the naming convention for projects.

Reply:
Nice, I like that more, I will rewrite it 👍

Reply:
So, I actually don't see the benefits there, would you mind to give a little bit of explaining? Why declare a struct such as [TemplateString](https://github.com/tuist/tuist/blob/master/Sources/ProjectDescription/TemplateString.swift) when all I need is custom `appendInterpolation` method? Seems to me like an unnecessary overhead.